src/main/java/org/mskcc/cbio/portal/scripts/UpdateCaseListsSampleIds.java

```diff
@@ -0,0 +1,242 @@
+ * FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ * is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+ * obligations to provide maintenance, support, updates, enhancements or
+ * modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+ * liable to any party for direct, indirect, special, incidental or
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.mskcc.cbio.portal.scripts;
+
+import joptsimple.OptionException;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import org.mskcc.cbio.portal.dao.DaoCancerStudy;
+import org.mskcc.cbio.portal.dao.DaoException;
+import org.mskcc.cbio.portal.dao.DaoSampleList;
+import org.mskcc.cbio.portal.model.CancerStudy;
+import org.mskcc.cbio.portal.model.SampleList;
+
+import java.io.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class UpdateCaseListsSampleIds extends ConsoleRunnable {
+
+    private File metaFile;
+    private File dataFile;
+    private List<File> caseListFiles = List.of();
+    private String cancerStudyStableId;
+    private Map<String, Set<String>> caseListSampleIdToSampleIds = new LinkedHashMap<>();
+    private DaoSampleList daoSampleList = new DaoSampleList();
+    private LinkedHashSet<String> allSampleIds;
+
+    public UpdateCaseListsSampleIds(String[] args) {
+        super(args);
+    }
+
+    /**
+     * Imports clinical data and clinical attributes (from the worksheet)
+     */
+    public void run() {
+        parseArguments();
+        readStudyIdAndDataFileFromMetaFile();
+        this.allSampleIds = readSampleIdsFromDataFile(this.dataFile);
+        // TODO has the all case list always to exist?
+        this.caseListSampleIdToSampleIds.put(cancerStudyStableId + "_all", this.allSampleIds);
+        Map<String, Set<String>> readCaseListSampleIds = readCaseListFiles();
+        this.caseListSampleIdToSampleIds.putAll(readCaseListSampleIds);
+        updateCaseLists(this.caseListSampleIdToSampleIds);
+    }
+
+    private Map<String, Set<String>> readCaseListFiles() {
```

Comment:
maybe similar code already exists elsewhere in the code?

Reply:
It does exist in `ImportSampleList.importSampleList(File dataFile)`. Although it's not reusable as it is now. Let me add TODO comments about that for now.

Reply:
Please see the following commit. I've extracted common logic to read and validate case list to separate classes https://github.com/se4bio/cbioportal-core/pull/1/commits/0b9c2c870597be55d8323058b85d25c57a0109a7